// antd-mobile/lib/list/style/index.native.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _reactNative = require('react-native');

var _default = require('../../style/themes/default.native');

var _default2 = _interopRequireDefault(_default);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

exports['default'] = {
    underlayColor: {
        backgroundColor: _default2['default'].fill_tap
    },
    Header: {
        fontSize: _default2['default'].font_size_base,
        color: _default2['default'].color_text_caption,
        paddingLeft: _default2['default'].h_spacing_lg,
        paddingRight: _default2['default'].h_spacing_lg,
        paddingTop: _default2['default'].v_spacing_lg,
        paddingBottom: _default2['default'].v_spacing_md,
        backgroundColor: _default2['default'].fill_body
    },
    Footer: {
        fontSize: _default2['default'].font_size_base,
        color: _default2['default'].color_text_caption,
        paddingLeft: _default2['default'].h_spacing_lg,
        paddingRight: _default2['default'].h_spacing_lg,
        paddingTop: _default2['default'].v_spacing_md,
        paddingBottom: _default2['default'].v_spacing_md,
        backgroundColor: _default2['default'].fill_body
    },
    Body: {
        backgroundColor: _default2['default'].fill_base,
        borderTopWidth: _reactNative.StyleSheet.hairlineWidth,
        borderTopColor: _default2['default'].border_color_base
    },
    BodyBottomLine: {
        position: 'absolute',
        bottom: 0,
        left: 0,
        right: 0,
        height: 1,
        backgroundColor: _default2['default'].fill_base,
        borderBottomWidth: _reactNative.StyleSheet.hairlineWidth,
        borderBottomColor: _default2['default'].border_color_base
    },
    Item: {
        flexGrow: 1,
        alignItems: 'center',
        flexDirection: 'row',
        paddingLeft: _default2['default'].h_spacing_lg,
        backgroundColor: _default2['default'].fill_base
    },
    Line: {
        flexDirection: 'row',
        flex: 1,
        alignItems: 'center',
        paddingRight: _default2['default'].h_spacing_lg,
        paddingTop: _default2['default'].v_spacing_sm,
        paddingBottom: _default2['default'].v_spacing_sm,
        minHeight: _default2['default'].list_item_height,
        borderBottomWidth: _reactNative.StyleSheet.hairlineWidth,
        borderBottomColor: _default2['default'].border_color_base
    },
    Thumb: {
        width: _default2['default'].icon_size_md,
        height: _default2['default'].icon_size_md,
        marginRight: _default2['default'].h_spacing_lg
    },
    Content: {
        color: _default2['default'].color_text_base,
        fontSize: _default2['default'].font_size_heading,
        lineHeight: Math.round(_default2['default'].font_size_heading * 1.1),
        textAlignVertical: 'center'
    },
    Extra: {
        color: _default2['default'].color_text_caption,
        fontSize: _default2['default'].font_size_heading,
        lineHeight: Math.round(_default2['default'].font_size_heading * 1.1),
        textAlign: 'right',
        textAlignVertical: 'center'
    },
    Brief: {
        minHeight: _default2['default'].font_size_icontext
    },
    BriefText: {
        color: _default2['default'].color_text_caption,
        fontSize: _default2['default'].font_size_subhead,
        paddingTop: _default2['default'].v_spacing_xs,
        textAlignVertical: 'center'
    },
    Arrow: {
        width: 8,
        height: 13,
        marginLeft: _default2['default'].h_spacing_md,
        marginTop: _default2['default'].v_spacing_xs
    },
    ArrowV: {
        width: 13,
        height: 8,
        marginLeft: _default2['default'].h_spacing_md
    },
    multipleLine: {
        paddingTop: _default2['default'].v_spacing_sm,
        paddingBottom: _default2['default'].v_spacing_sm
    },
    multipleThumb: {
        width: _default2['default'].icon_size_lg,
        height: _default2['default'].icon_size_lg
    },
    column: {
        flex: 1,
        flexDirection: 'column'
    }
};
module.exports = exports['default'];
