// antd-mobile/lib/action-sheet/demo/basic.native.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.description = exports.title = undefined;

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactNative = require('react-native');

var _antdMobile = require('antd-mobile');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

var Test = function (_React$Component) {
    (0, _inherits3['default'])(Test, _React$Component);

    function Test() {
        (0, _classCallCheck3['default'])(this, Test);

        var _this = (0, _possibleConstructorReturn3['default'])(this, (Test.__proto__ || Object.getPrototypeOf(Test)).call(this));

        _this.showActionSheet = function () {
            var BUTTONS = ['Operation1', 'Operation2', 'Operation3', 'Delete', 'Cancel'];
            _antdMobile.ActionSheet.showActionSheetWithOptions({
                title: 'Title',
                message: 'Description',
                options: BUTTONS,
                cancelButtonIndex: 4,
                destructiveButtonIndex: 3
            }, function (buttonIndex) {
                _this.setState({ clicked: BUTTONS[buttonIndex] });
            });
        };
        _this.showShareActionSheet = function () {
            var opts = {
                message: 'Message to go with the shared url',
                title: 'Share Actionsheet'
            };
            if (_reactNative.Platform.OS === 'ios') {
                opts.url = 'https://www.alipay.com/';
                opts.tintColor = '#ff0000';
                opts.excludedActivityTypes = ['com.apple.UIKit.activity.PostToTwitter'];
            }
            _antdMobile.ActionSheet.showShareActionSheetWithOptions(opts, function (error) {
                return alert(error);
            }, function (success, method) {
                var text = void 0;
                if (success) {
                    text = 'Shared with ' + method;
                } else {
                    text = 'Did not share';
                }
                _this.setState({ text: text });
            });
        };
        _this.state = {
            clicked: 'none',
            text: ''
        };
        return _this;
    }

    (0, _createClass3['default'])(Test, [{
        key: 'render',
        value: function render() {
            return _react2['default'].createElement(
                _reactNative.View,
                { style: { marginTop: 30 } },
                _react2['default'].createElement(
                    _reactNative.View,
                    { style: [{ padding: 8 }] },
                    _react2['default'].createElement(
                        _antdMobile.Button,
                        { onClick: this.showActionSheet },
                        'showActionSheet'
                    )
                ),
                _react2['default'].createElement(
                    _reactNative.Text,
                    { style: [{ padding: 8 }] },
                    'clicked button: ',
                    this.state.clicked
                ),
                _react2['default'].createElement(
                    _reactNative.View,
                    { style: [{ padding: 8 }] },
                    _react2['default'].createElement(
                        _antdMobile.Button,
                        { onClick: this.showShareActionSheet },
                        'showShareActionSheet'
                    )
                ),
                _react2['default'].createElement(
                    _reactNative.Text,
                    { style: [{ padding: 8 }] },
                    this.state.text
                )
            );
        }
    }]);
    return Test;
}(_react2['default'].Component);

exports['default'] = Test;
var title = exports.title = 'ActionSheet';
var description = exports.description = 'ActionSheet example';
